### PR TITLE
Add nick to automatically generated staging name

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 

the `--auto`option in the `biomage stage` command is very helpful in generating names automaticallly. However, the generated names does not contain the name of the user that generates the staging environment, making it difficult to see in Lens who created that staging environment with the automatically generated name.

This PR adds the username / `BIOMAGE_NICK` to the automatically generated name.

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] New commands have been added to Makefile/test
- [ ] Unit tests written
- [ ] Run make test
- [ ] Test any modified command

To set up easy local testing with inframock, follow the instructions here: https://github.com/hms-dbmi-cellenics/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/hms-dbmi-cellenics/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: biomage/stage/stage.py [hunk 1]
```diff
@@ -202,13 +202,14 @@ def get_sandbox_id(templates, manifests, all_experiments, auto):
             if opts.ref != DEFAULT_BRANCH
         ]
     )
+    user_name = re.sub(r"[^\w\s]", "", os.getenv("BIOMAGE_NICK", os.getenv("USER", "")))
 
     # if we are in auto mode (non interactive) just generate a random sandbox ID name
     if auto:
-        return random_name.generate_name()
+        return (user_name + "-" + random_name.generate_name()).lower()[:26]
```
Comment (kafkasl) [inline biomage/stage/stage.py:209]: I think you can't cut with `[:26]` because if it ends in a `-` then it's gonna be an invalid name iirc

Comment (kafkasl) [reply biomage/stage/stage.py:209]: I recall something like that, but it does not make a lot of sense. Maybe just give it a try

Comment (aerlaut) [reply biomage/stage/stage.py:209]: You're right, Kubernetes fails to create the namespace : 

```
2022-03-25 09:55:18 [ℹ] eksctl version 0.88.0 2022-03-25 09:55:18 [ℹ] using region eu-west-1 2022-03-25 09:55:21 [ℹ] creating Fargate profile "worker-agi-" on EKS cluster "biomage-staging" Error: could not create fargate profiles: failed to create Fargate profile "worker-agi-" on EKS cluster "biomage-staging": failed to create Fargate profile "worker-agi-": InvalidParameterException: The namespace provided is invalid. { RespMetadata: { StatusCode: 400, RequestID: "29dcfa60-03df-4150-a18f-b9fdf9944b88" }, Message_: "The namespace provided is invalid." }
```

Thanks for the heads up Pol! I'll add a check to ensure that the last element isn't a dash

Comment (aerlaut) [reply biomage/stage/stage.py:209]: As explained in detail : https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names

TLDR Kubernetes object names must : 

- contain no more than 253 characters
- contain only lowercase alphanumeric characters, '-' or '.'
- start with an alphanumeric character
- end with an alphanumeric character

**Review by kafkasl — APPROVED** [review]
